### DISCUSSION TITLE
Reading from serial with dut.expect()

### DISCUSSION THREAD

**Discussion by KaDw** [espressif/pytest-embedded#270] [Q&A] [answered]
Hi,

Is there a way to get data full data from the serial after dut.write and before the dut.expect?
I expect the following from the dut
```
adcval=2.33V
Operation successful
```
Here is the test, now besides expecting the Operation successful I need the `adcval=2.33V` line to check if it is within the range
```
dut.write("getadc")
ret = dut.expect("Operation successful")
print(ret.group()) # this gives me only b"Operation successful"
# parse adcval=2.33V and verify if it is within the range
```

**Comment by igrr** [comment, marked as answer]
Perhaps you could do
```python
match = dut.expect(r'adcval=([0-9.]+)V\nOperation successful')
value = match.group(1)
```

(I didn't check whether this regex works for your output, but hopefully this illustrates the idea.)

**Comment by KaDw** [comment]
Okay, I didn't look at it this way. I see the point, thanks for the quick reply!
